core/app/models/workarea/bulk_action/product_edit/price_change/amount.rb

```diff
@@ -0,0 +1,80 @@
+module Workarea
+  class BulkAction
+    class ProductEdit < BulkAction
+      class PriceChange
+        # Calculates the new amount of a given current price using
+        # directives from the Bulk Product Edit form.
+        class Amount
+          class InvalidError < RuntimeError
+            def initialize(amount)
+              super "Invalid Amount From #{amount.inspect}"
+            end
+          end
+
+          attr_reader :current, :type, :action, :amount
+
+          # @param [Money] current - Current price
+          # @param [String] action - Either "set", "increase", or "decrease"
+          # @param [String] type - Either "flat" or "percentage"
+          # @param [String] amount - Amount to increase/set/decrease by
+          def initialize(current, action:, type:, amount: 0.to_m)
+            @current = current || 0.to_m
+            @action = action
+            @type = type
+            @amount = amount
+          end
+
+          # Test whether this amount is not setting a flat amount. Used
+          # in the `PriceChange` class when determining whether to
+          # perform an update. Non-generic prices are not updated when
+          # the admin chooses to set a flat amount for all prices in the
+          # product's SKUs.
+          #
+          # @return [Boolean] Whether the admin wants to set a flat amount.
+          def multiplier?
+            return true if action != 'set'
+
+            type != 'flat'
```

Comment:
These 2 `!=` are kind of confusing, aren't there specific values to compare against to make it a little clearer?

Reply:
Not that I can tell. A price change is a "multiplier" if you're not trying to set a flat amount.